browse/services/database/__init__.py

```diff
@@ -0,0 +1,154 @@
+"""Import db instance and define utility functions."""
+
+import ipaddress
+from datetime import datetime
+from dateutil.tz import tzutc, gettz
+from typing import List, Optional, Any, Callable
+from sqlalchemy.sql import func
+from sqlalchemy.orm import Query
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy.exc import OperationalError, DBAPIError
+from arxiv.base.globals import get_application_config
+
+from browse.services.database.models import db, Document, \
+    MemberInstitution, MemberInstitutionIP, TrackbackPing, SciencewisePing, \
+    DBLP, DBLPAuthor, DBLPDocumentAuthor
+from arxiv.base import logging
+from logging import Logger
+
+logger = logging.getLogger(__name__)
+app_config = get_application_config()
+
+
+def db_handle_error(logger: Logger, default_return_val: Any) \
```

Comment:
Should something like this go into `arxiv.base`?

Reply:
db_handler_error? It could if developers wanted it. Unlike arxiv IDs and categories, there currently isn't a plan to move utility code like this to arxiv-base.